pkg/helm/controller/reconcile_test.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func TestHasHelmUpgradeForceAnnotation(t *testing.T) {
+	tests := []struct {
+		input       map[string]interface{}
+		expectedVal bool
+		expectedOut string
+		name        string
+	}{
+		{
+			input: map[string]interface{}{
+				"helm.operator-sdk/upgrade-force": "True",
+			},
+			expectedVal: true,
+			name:        "base case true",
+		},
+		{
+			input: map[string]interface{}{
+				"helm.operator-sdk/upgrade-force": "False",
+			},
+			expectedVal: false,
+			name:        "base case false",
+		},
+		{
+			input: map[string]interface{}{
+				"helm.operator-sdk/upgrade-force": "1",
+			},
+			expectedVal: true,
+			name:        "true as int",
+		},
+		{
+			input: map[string]interface{}{
+				"helm.operator-sdk/upgrade-force": "0",
+			},
+			expectedVal: false,
+			name:        "false as int",
+		},
+		{
+			input: map[string]interface{}{
+				"helm.operator-sdk/wrong-annotation": "true",
+			},
+			expectedVal: false,
+			name:        "annotation not set",
+		},
+		{
+			input: map[string]interface{}{
+				"helm.operator-sdk/upgrade-force": "invalid",
+			},
+			expectedVal: false,
+			name:        "invalid value",
+		},
+	}
+
+	for _, test := range tests {
+		assert.Equal(t, test.expectedVal, hasHelmUpgradeForceAnnotation(annotations(test.input)), test.name)
```

Comment:
Would not be required used assert in this case. We could do something such as follows but I am ok with 👍 

```suggestion
		got := hasHelmUpgradeForceAnnotation(annotations(test.input))
		if test.expectedVal != got {
		       t.Fatalf("For hasHelmUpgradeForceAnnotation expected the value %v and got %v", test.expectedVal, got)
		}
```

Reply:
Is assert bad? I much prefer asserts

Reply:
I am ok with move forward with and merge with the asserts. 

However, according to the Golang authors they have reasons to do not added it implicitly in the language. Also, for example, here using the Testing golang only we do not need to print the name of the test it will be done by default.

Reply:
Please, feel free to merge this one if you are also ok with @jmrodri .

Reply:
I prefer asserts and think that some of the opinions in the go FAQ are just that, opinions.

Unless we as a community are _extremely_ diligent about test style using the standard testing library:
1. We end up with a total hodge podge of not-so-great tests.
2. The failure messages are inconsistent at best and not helpful at worst.
3. Writing tests take longer and more lines of code, which means time not spent adding more features or fixing more bugs AND it is more to review, understand, and maintain.

Regardless, I don't want to start a flame war 😆. We're using `github.com/stretchr/testify` in many (perhaps most) of our tests already, and I think consistency is the most important thing here.